backend/app/config_upload/config_upload_service.py

```diff
@@ -0,0 +1,31 @@
+from .netmiko_connection_config import NetmikoConnectionConfig
+from ..running_config.util.device_config_types import DeviceConfigInfo
+from .util.netmiko_device import NetmikoDevice
+from .exceptions.config_upload_exceptions import DeviceConfigError, DeviceConnectionError
+
+
+class ConfigUploadService:
+    def build_netmiko_devices(self, topology_config: list[DeviceConfigInfo]):
+        devices: list[NetmikoDevice] = []
+        netmiko_connection_config = NetmikoConnectionConfig()
+        for device_config_info in topology_config:
+            netmiko_device = NetmikoDevice(
+                device_type=netmiko_connection_config.get_device_type(),
+                host=netmiko_connection_config.get_host(),
+                username=netmiko_connection_config.get_username(),
+                password=netmiko_connection_config.get_password(),
+                config=netmiko_connection_config.get_config(device_config_info['dev_running_config'])
+            )
+            devices.append(netmiko_device)
+        return devices
+
+    def upload_configs(self, devices: list[NetmikoDevice]):
+        for device in devices:
+            try:
+                with device:
+                    try:
+                        device.send_config_commands()
+                    except Exception as e:
+                        raise DeviceConfigError(f"Failed to send config to {device.host}. Error:{e}")
+            except Exception as e:
+                raise DeviceConnectionError(f"Failed to connect to {device.host}. Error:{e}")
```

Comment:
Tutaj nie lepszym pomysłem byłoby zrobienie error handlingu tak jak np w linijkach 123-129 w `upload_config_router.py`? Mam na myśli skorzystanie z jednego bloku `try` i dwóch `except` na tym samym poziomie.

No i ten ogólny wyjątek `Exception` trochę mi nie siedzi. Jeśli wiemy jakie wyjątki mogą rzucić te metody (a podejrzewam, że w dokumentacji jest to opisane - popraw mnie jeśli jednak nie) to starajmy się łapać te konkretne wyjątki, a dopiero na sam koniec jeśli nie mamy pewności czy to wszystkie to zrobić takiego `excepta` z łapaniem ogólnego wyjątku.